pyqtorch/modules/parametric.py

```diff
@@ -22,6 +22,51 @@ def __init__(self, gate: str, qubits: ArrayLike, n_qubits: int):
         self.register_buffer("imat", OPERATIONS_DICT["I"])
         self.register_buffer("paulimat", OPERATIONS_DICT[gate])
 
+    def __mul__(self, other: RotationGate | QuantumCircuit) -> QuantumCircuit:
+        if self.n_qubits != other.n_qubits:
+            raise ValueError(
+                (
+                    f"Number of Qubits don't match. "
+                    f"Left gate is applied on a {self.n_qubits} qubit system and "
+                    f"right gate is applied on a {other.n_qubits} qubit system."
+                )
+            )
+
+        if isinstance(other, RotationGate):
+            return QuantumCircuit(self.n_qubits, [self, other])
+
+        if isinstance(other, QuantumCircuit):
+            return QuantumCircuit(self.n_qubits, other.operations.insert(0, self))
+
+        else:
+            return NotImplemented
+
+    def __rmul__(self, other: QuantumCircuit) -> QuantumCircuit:
```

Comment:
I think this one should go into the `QuantumCircuit` itself, see one of the comments above about `circuit*gate`